tequilapi/endpoints/service.go

```diff
@@ -261,9 +246,53 @@ func AddRoutesForService(router *httprouter.Router, serviceManager ServiceManage
 	router.DELETE("/services/:id", serviceEndpoint.ServiceStop)
 }
 
-func toServiceRequest(req *http.Request) (sr serviceRequest, err error) {
-	err = json.NewDecoder(req.Body).Decode(&sr)
-	return sr, err
+func (se *ServiceEndpoint) toServiceRequest(req *http.Request) (serviceRequest, error) {
+	var jsonData struct {
+		ProviderID  string           `json:"providerId"`
+		ServiceType string           `json:"serviceType"`
+		Options     *json.RawMessage `json:"options"`
+	}
+	if err := json.NewDecoder(req.Body).Decode(&jsonData); err != nil {
+		return serviceRequest{}, err
+	}
+
+	sr := serviceRequest{
+		ProviderID:  jsonData.ProviderID,
+		ServiceType: se.toServiceType(jsonData.ServiceType),
+		Options:     se.toServiceOptions(jsonData.ServiceType, jsonData.Options),
+	}
+	return sr, nil
+}
+
+func (se *ServiceEndpoint) toServiceType(value string) string {
+	if value == "" {
+		return ""
```

Comment:
Is empty string a valid service type? Shouldn't we return `serviceTypeInvalid`?

Reply:
No, empty value is non given value, and we dont event want to parse it with further logic in function